eclair-core/src/main/scala/fr/acinq/eclair/NodeParams.scala

```diff
@@ -118,18 +120,49 @@ object NodeParams {
       .withFallback(ConfigFactory.parseFile(new File(datadir, "eclair.conf")))
       .withFallback(ConfigFactory.load())
 
-  def getSeed(datadir: File): ByteVector = {
-    val seedPath = new File(datadir, "seed.dat")
-    if (seedPath.exists()) {
-      ByteVector(Files.readAllBytes(seedPath.toPath))
-    } else {
-      datadir.mkdirs()
-      val seed = randomBytes32
-      Files.write(seedPath.toPath, seed.toArray)
-      seed
+  private def readSeedFromFile(seedPath: File): ByteVector = {
+    logger.info(s"use seed file: ${seedPath.getCanonicalPath}")
+    ByteVector(Files.readAllBytes(seedPath.toPath))
+  }
+
+  private def writeSeedToFile(path: File, seed: ByteVector): Unit = {
+    Files.write(path.toPath, seed.toArray)
+    logger.info(s"create new seed file: ${path.getCanonicalPath}")
+  }
+
+  private def migrateSeedFile(old: File, `new`: File): Unit = {
```

Comment:
I think the migration is misleading. We cannot really migrate an existing node, so maybe we should instead be explicit about that?

Using the same seed for node and channel means we didn't segregate anything and if the user moves his `channelSeed.dat` to a hardware wallet, he will get a false sense of security.

It also won't make sense to generate the channel seed on a normal machine, it should be generated by a hardware wallet, shouldn't it?

Reply:
Yes but reusing an existing seed that was used on a regular computer (moving i into to a hardware wallet for example) is not safe anyway ?

Reply:
I agree, maybe it's the term "migrating" that I find misleading there, but it may make sense here...
I need to think about this more.

Reply:
Ok so what we're doing for now is migrating nodes from a single seed to two seeds, so the wording makes sense.
I guess later we'll provide tutorials and additional changes to initialize a node with a hardware device that handles all the `channelSeed` part.